### ISSUE TITLE
Please add Nothing<T>

### ISSUE THREAD

**Issue by bsimser** [gregoryyoung/nothing#24]

**Comment by dsyme** [comment]
Yes, this is needed, obviously. It's a shame the original designers didn't consider this direction when they got started.

I suggest we also constrain `T` to be the underlying representation of Nothingness, `Nothing<TRepr>`.

Here `TRepr` can be instantiated to `Platonic`, `ZenBuddhist`, `Nihilist`, `Mathematical` and the infinitely-recursive `Nothing<Nothing<Nothing<Nothing<.....>>>>` that can only be written in languages supporting infinite nothing types.
